torch/_meta_registrations.py

```diff
@@ -143,3 +133,99 @@ def meta_repeat_interleave_Tensor(repeats, output_size=None):
             "cannot repeat_interleave a meta tensor without output_size"
         )
     return repeats.new_empty(output_size)
+
+@torch.library.impl(meta_lib, "index.Tensor")
+def meta_index_Tensor(self, indices):
```

Comment:
If this function can be called from anywhere other than the `operator[]` in Python, it should probably also check that `indices` is not an empty list.